File: 2-week/set/readability.c
#include <cs50.h>
#include <stdio.h>

static void handler_text(string txt, int *qtd_ltt, int *qtd_stc, int *qtd_wrd);

int main(void)
{
    int qtd_ltt = 0;
    int qtd_stc = 0;
    int qtd_wrd = 0;
    // string text = get_string("Text: ");
    string text = "Congratulations! Today is your day. You're off to Great Places! You're off and away!";

    handler_text(text, &qtd_ltt, &qtd_stc, &qtd_wrd);

    printf("Quantidade de letra: %i", qtd_ltt);

}

static void handler_text(string txt, int *qtd_ltt, int *qtd_stc, int *qtd_wrd)
{
    int i = 0;
    while (txt[i])
    {
        // Simulate isAlpha func.
        if((txt[i] >= 'a' && txt[i] <= 'z') || (txt[i] >= 'A' && txt[i] <= 'Z'))
        {
            // +1 Letter
            *qtd_ltt = *qtd_ltt + 1;
        }
        else if (txt[i] == ' ')
        {
            // +1 word
            *qtd_wrd = *qtd_wrd + 1;
        }
        // Check end of sentence.
        else if(txt[i] == '!' || txt[i] == '.' || txt[i] == '?'
                || txt[i] == ':')
        {
            // Handle '...'. Just one sentence. Increment 'i' variable.
            if ((txt[i+1] && txt[i+2]) && (txt[i] == '.' && txt[i+1] == '.'
                    && txt[i+2] == '.'))
            {
                i = i + 2;
            }
            // +1 Sentence
            *qtd_stc = *qtd_stc + 1;
        }
        i++;
    }
}

File: 6-week/problem/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")

    qtd_ltt, qtd_stc, qtd_wrd = handle_text(text)
    handle_calc(qtd_ltt, qtd_stc, qtd_wrd)


def handle_text(txt):

    # Initialize variables
    qtd_wrd = 0
    qtd_ltt = 0
    qtd_stc = 0

    # word
    qtd_wrd = len(txt.split())

    # Verify Sentence
    for c in txt:
        if c == "." or c == "!" or c == "?":
            qtd_stc += 1
        elif c.isalpha():
            qtd_ltt += 1

    # Return results
    return qtd_ltt, qtd_stc, qtd_wrd


def handle_calc(qtd_ltt, qtd_stc, qtd_wrd):
    letter_per_hundred_words = float(qtd_ltt / qtd_wrd * 100)
    sentence_per_hundred_words = float(qtd_stc / qtd_wrd * 100)

    result = float(0.0588 * letter_per_hundred_words - 0.296 * sentence_per_hundred_words - 15.8)
    int_result = round(result)

    if int_result >= 16:
        print("Grade 16+")
    elif int_result < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {int_result}")


if __name__ == "__main__":
    main()
